File: notif-ingester/models/account.py
# Source: c/Models/Account.cs
from typing import Optional
from pydantic import BaseModel, Field
from models.enums import AccountType

class Account(BaseModel):
    id: str = Field(alias="Id")
    user_id: str = Field(alias="UserId")
    account_group_id: str = Field(alias="AccountGroupId")
    name: str = Field(alias="Name")
    starting_balance: float = Field(alias="StartingBalance")
    current_balance: float = Field(alias="CurrentBalance")
    account_type: AccountType = Field(alias="AccountType")
    description: Optional[str] = Field(default=None, alias="Description")
    tags: Optional[list[str]] = Field(default_factory=list, alias="Tags")
    credit_card_cycle_start_day: Optional[int] = Field(default=None, alias="CreditCardCycleStartDay")
    credit_card_payment_due_day: Optional[int] = Field(default=None, alias="CreditCardPaymentDueDay")

    class Config:
        populate_by_name = True

File: backend/Models/Account.cs
// Source: C# (Original)
using System.Text.Json.Serialization;

namespace FinanceApp.Models
{
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public enum AccountType
    {
        Cash,
        Bank,
        CreditCard,
        Investment,
        Asset,
        Liability,
        Equity,
        Income,
        Expense,
        Adjustment
    }

    public class Account
    {
        public string Id { get; set; } = Guid.CreateVersion7().ToString();
        public string UserId { get; set; } = string.Empty;
        public string AccountGroupId { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        private List<string>? _tags = new();
        public List<string> Tags 
        { 
            get => _tags ??= new(); 
            set => _tags = value ?? new(); 
        }
        public decimal StartingBalance { get; set; }
        public decimal CurrentBalance { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public AccountType AccountType { get; set; }
        public int? CreditCardCycleStartDay { get; set; }
        public int? CreditCardPaymentDueDay { get; set; }
    }
}
